client/src/App.test.jsx

```diff
@@ -15,169 +15,193 @@ import { Router } from 'react-router-dom'
 
 it('renders correctly', () => {
   const tree = renderer
-    .create(<App />)
+    .create(<AppsRoot />)
     .toJSON();
   expect(tree).toMatchSnapshot();
 });
 
-test('Scroll to next month button', async() => {
-  render(<App />)
-  const marchMonthsBefore = await screen.queryByText('March');
-  expect(marchMonthsBefore).toBe(null);
-
-  fireEvent.click(screen.getByText('Check Availability'))
-  fireEvent.click(screen.getByText('>>'));
-
-  const marchMonthsAfter = await screen.findAllByText('March');
-
-
-  expect(marchMonthsAfter).toHaveLength(1);
-})
-
 
 const server = setupServer(
   rest.get('/109/availableDates', (req, res, ctx) => {
-
     return res(ctx.json(exampleData));
   }),
   rest.get('/109/minNightlyRate', (req, res, ctx) => {
     return res(ctx.json({minNightlyRate: 888}))
   })
 )
 
-beforeAll(() => server.listen())
-afterEach(() => server.resetHandlers())
-afterAll(() => server.close())
+beforeAll(() => server.listen());
 
+beforeEach( () => {
+  console.log('> setup test')
+});
 
+afterEach( () => {
+  console.log('< teardown test')
+  server.resetHandlers();
+});
 
-test('Calendar has different formatting for available, unavailable, and checkOutOnly dates', async() => {
-  render(<App />)
+afterAll(() => server.close());
 
+describe("Calendar tests", () => {
 
-  fireEvent.click(screen.getByText('Check Availability'));
 
-  const blockedDates = await screen.findAllByTestId('blocked');
-  const normalDates = await screen.findAllByTestId('normal');
-  const checkOutOnly = await screen.findAllByTestId('checkOutOnly');
-  expect(blockedDates[0]).toBeInTheDocument();
-  expect(normalDates[0]).toBeInTheDocument();
-  expect(checkOutOnly[0]).toBeInTheDocument();
 
-})
 
-test('Hovering over an available date changes its formatting to \'hovered date\'', async() => {
-  render(<App />)
 
 
-  fireEvent.click(screen.getByText('Check Availability'));
-  const normalDates = await screen.findAllByTestId('normal');
+  test('Scroll to next month button', async() => {
+    const { unmount } = render(<AppsRoot />)
+    const marchMonthsBefore = await screen.queryByText('March');
+    expect(marchMonthsBefore).toBe(null);
 
-  fireEvent.mouseOver(normalDates[0]);
+    fireEvent.click(screen.getByText('Check Availability'))
+    fireEvent.click(screen.getAllByText('>>')[0]);
 
-  const blockedDates = await screen.findAllByTestId('hoveredDate');
-  expect(blockedDates[0]).toBeInTheDocument();
+    const marchMonthsAfter = await screen.findAllByText('March');
 
 
-})
+    expect(marchMonthsAfter).toHaveLength(1);
+    unmount();
+  })
 
-test('Selecting a check-in date updates the listed check-in date and changes the date style', async() => {
-  render(<App />)
 
+  test('Calendar has different formatting for available, unavailable, and checkOutOnly dates', async() => {
+    const { unmount } = render(<AppsRoot />)
```

Comment:
Is it possible to have your render and unmount happen in a beforeEach()/afterEach()? I haven't dug into this yet, but it seems to be repeated in pretty much every test. If so it could keep code more DRY.